### DISCUSSION TITLE
MLMG always reports: Initial rhs = 0 when I pass a vector of MultiFab rhs with a constant value.

### DISCUSSION THREAD

**Discussion by ZiyangHuang** [AMReX-Codes/amrex#3507] [Q&A] [answered]
Greetings,

When I used MLMG to solve ABecLaplacian, I gave a constant ```rhs``` via ```rhs[lev].setVal(val)``` for all ```lev```, where ```rhs``` is a vector of ```MultiFab```. No matter what ```val``` I input to ```setVal()``` (I tried 6.0 and 100.0), MLMG always reported: Initial rhs = 0, and no iteration was performed. However, if ```rhs``` is not a constant, then MLMG reported a non-zero initial rhs. I have no idea what causes the issue.

**Comment by zingale** [comment]
Are you periodic on all boundaries?

**Reply by WeiqunZhang** [reply]
periodic or Neumann

**Reply by ZiyangHuang** [reply]
Yes, the boundary conditions are either periodic or homogeneous Neumann.

**Reply by zingale** [reply, marked as answer]
In that case, solvability requires that the source integrate to zero over the domain, so I imagine that the mlmg is subtracting this off of the source before doing the solve.

**Reply by ZiyangHuang** [reply]
I see, thanks for your reply.

**Reply by WeiqunZhang** [reply]
assuming the A is zero in your ABecLaplacian

**Reply by ZiyangHuang** [reply]
Yes, ascalar and acoef are zero.
